test/test_torch.py

```diff
@@ -3068,23 +3069,106 @@ def test_Size(self):
         self.assertIsInstance(x[:-1], torch.Size)
         self.assertIsInstance(x + x, torch.Size)
 
-    def test_transpose_neg(self):
-        x = torch.randn(10, 20, 30)
-        ndim = 3
 
-        for i, j in combinations(range(ndim), 2):
-            a = x.transpose(i, j)
-            b = x.transpose(i - ndim, j - ndim)
-            self.assertEqual(a, b)
-
-            a = torch.transpose(x, i, j)
-            b = torch.transpose(x, i - ndim, j - ndim)
-            self.assertEqual(a, b)
-
-            a = x.clone()
-            x.transpose_(i, j)
-            x.transpose_(i - ndim, j - ndim)
-            self.assertEqual(a, x)
+def compare_pos_neg_dim(name, tensor_arg, arg_constr, types):
+    def tmp(self):
+        if isinstance(tensor_arg, list):
+            assert 0 not in types and 1 not in types
+            x = []
+            for arg in tensor_arg:
+                x.append(torch.randn(*arg))
+                ndim = len(arg)
+        else:
+            x = torch.randn(*tensor_arg)
+            ndim = len(tensor_arg)
+
+        n_dim_to_test = 0
+        tmp_arg = arg_constr()
+        for el in tmp_arg:
+            if el is None:
+                n_dim_to_test += 1
+
+        for dims_val in combinations(range(ndim), n_dim_to_test):
+            arg = arg_constr()
+            arg_neg = copy.copy(arg)
```

Comment:
The problem is that if there is a random vector in the arguments, they will be different if I call `arg_constr()` again.